java/com/google/flatbuffers/Table.java

```diff
@@ -69,7 +69,8 @@ protected int __offset(int vtable_offset) {
 
   protected static int __offset(int vtable_offset, int offset, ByteBuffer bb) {
     int vtable = bb.array().length - offset;
```

Comment:
This will break for many different type of byte buffers.  Both heap and off-heap.

- If the byte buffer is a slice and on-heap, this will fail because ``arrayOffset`` needs to be added.  
- If the byte buffer is off-heap this will also throw

Using ``limit`` might not be correct either depending on the context.